rbc/tests/test_omnisci_udtf.py

```diff
@@ -1,49 +1,140 @@
-              + '.'.join(map(str, available_version)))
-    available_version = ()
-pytestmark = pytest.mark.skipif(not available_version, reason=reason)
+from rbc.tests import omnisci_fixture
+import numpy as np
 
 
 @pytest.fixture(scope='module')
 def omnisci():
-    config = rbc_omnisci.get_client_config(debug=not True)
-    m = rbc_omnisci.RemoteOmnisci(**config)
-    table_name = 'rbc_test_omnisci_udtf'
-    m.sql_execute('DROP TABLE IF EXISTS {table_name}'.format(**locals()))
-
-    sqltypes = ['FLOAT', 'DOUBLE', 'TINYINT', 'SMALLINT', 'INT', 'BIGINT',
-                'BOOLEAN']
-    colnames = ['f4', 'f8', 'i1', 'i2', 'i4', 'i8', 'b']
-    table_defn = ',\n'.join('%s %s' % (n, t)
-                            for t, n in zip(sqltypes, colnames))
-    m.sql_execute(
-        'CREATE TABLE IF NOT EXISTS {table_name} ({table_defn});'
-        .format(**locals()))
+    for o in omnisci_fixture(globals()):
+        define(o)
+        yield o
+
+
+def define(omnisci):
+
+    T = ['int64', 'int32', 'int16', 'int8', 'float64', 'float32']
+
+    @omnisci('int32(Cursor<int64, T>, T, RowMultiplier,'
+             ' OutputColumn<int64>, OutputColumn<T>)', T=T)
+    def sqlmultiply(rowid, col, alpha, row_multiplier, rowid_out, out):
+        for i in range(len(col)):
+            j = rowid[i]
+            out[j] = col[i] * alpha
+            rowid_out[j] = i
+        return len(col)
+
+    @omnisci('int32(Cursor<int64, T>, T, RowMultiplier,'
+             ' OutputColumn<int64>, OutputColumn<T>)', T=T)
+    def sqladd(rowid, col, alpha, row_multiplier, rowid_out, out):
+        for i in range(len(col)):
+            j = rowid[i]
+            out[j] = col[i] + alpha
+            rowid_out[j] = i
+        return len(col)
+
+    @omnisci('int32(Cursor<int64, T>, Cursor<int64, T>, RowMultiplier,'
+             ' OutputColumn<int64>, OutputColumn<T>)', T=T)
+    def sqladd2(rowid1, col1, rowid2, col2, row_multiplier, rowid_out, out):
+        for i1 in range(len(col1)):
+            j1 = rowid1[i1]
+            for i2 in range(len(col2)):
+                j2 = rowid2[i2]
+                if j1 == j2:
+                    out[j1] = col1[i1] + col2[i2]
+                    rowid_out[j1] = i1
+                    break
+        return len(col1)
+
+    @omnisci('int32(Cursor<int64, T>, Cursor<int64, T>, RowMultiplier,'
+             ' OutputColumn<int64>, OutputColumn<T>)', T=T)
+    def sqlmultiply2(rowid1, col1, rowid2, col2, row_multiplier, rowid_out, out):
+        for i1 in range(len(col1)):
+            j1 = rowid1[i1]
+            for i2 in range(len(col2)):
+                j2 = rowid2[i2]
+                if j1 == j2:
+                    out[j1] = col1[i1] * col2[i2]
+                    rowid_out[j1] = i1
+                    break
+        return len(col1)
+
+
+@pytest.mark.parametrize("kind", ['i8', 'i4', 'i2', 'i1', 'f8', 'f4'])
+def test_composition(omnisci, kind):
+    omnisci.require_version((5, 6), 'Requires omniscidb-internal PR 5440')
```

Comment:
```suggestion
    omnisci.require_version((5, 7), 'Requires omniscidb-internal PR 5440')
```
The omniscidb-internal version is 5.7, but this always confuses me.

Reply:
yeah, it seems the omniscidb-internal master is always ahead by 0.1 of an upcoming release version.